plugins/in_systemd/systemd.c

```diff
@@ -333,6 +352,12 @@ static int in_systemd_collect_archive(struct flb_input_instance *i_ins,
         return 0;
     }
 
+    if (ret == FLB_SYSTEMD_ERROR) {
```

Comment:
where the systemd context is being re-created ?, how is this part being tested ?

Reply:
The context is being re-created somewhere in retry loop.
I've removed this code. It's not needed.